CMakeLists.txt

```diff
@@ -140,75 +146,142 @@ if("${isSystemDir}" STREQUAL "-1")
+			GIT_TAG        v2021.11.0
+			GIT_PROGRESS   TRUE
+			)
+		FetchContent_MakeAvailable(TBB)
+		set_property(DIRECTORY ${tbb_SOURCE_DIR} PROPERTY EXCLUDE_FROM_ALL YES)
+		# note: we do want to install tbb to the user machine when built from
+		# source
+		if(NOT EMSCRIPTEN)
+			install(TARGETS tbb)
+		endif()
+	endif()
+	message("Parallel Backend: ${MANIFOLD_PAR}")
+elseif(NOT MANIFOLD_PAR OR MANIFOLD_PAR STREQUAL "NONE")
+	message("Parallel Backend: NONE")
+	set(MANIFOLD_PAR "NONE")
+else()
+	message(FATAL_ERROR "Invalid value for MANIFOLD_PAR: ${MANIFOLD_PAR}. "
+		"Should be \"TBB\" or \"NONE\"")
+endif()
 
+# If we're building cross_section, we need Clipper2
 if(MANIFOLD_CROSS_SECTION)
-  set(TEMPLATE_OPTIONAL_CLIPPER "Clipper2")
+	find_package(Clipper2 QUIET)
+	if(NOT Clipper2_FOUND AND PKG_CONFIG_FOUND)
+		pkg_check_modules(Clipper2 Clipper2)
+	endif()
+	if(Clipper2_FOUND)
+		add_library(Clipper2 SHARED IMPORTED)
+		set_property(TARGET Clipper2 PROPERTY
+			IMPORTED_LOCATION ${Clipper2_LINK_LIBRARIES})
+		if(WIN32)
+			set_property(TARGET Clipper2 PROPERTY
+				IMPORTED_IMPLIB ${Clipper2_LINK_LIBRARIES})
+		endif()
+		target_include_directories(Clipper2 INTERFACE ${Clipper2_INCLUDE_DIRS})
+	else()
+		if (NOT MANIFOLD_DOWNLOADS)
+			message(FATAL_ERROR "cross_section enabled, but Clipper2 was not found and dependency downloading is disabled.  Please install Clipper2 and reconfigure.\n")
+		endif()
+		message(STATUS "clipper2 not found, downloading from source")
+		set(CLIPPER2_UTILS OFF)
+		set(CLIPPER2_EXAMPLES OFF)
+		set(CLIPPER2_TESTS OFF)
+		set(CLIPPER2_USINGZ "OFF" CACHE STRING "Preempt cache default of USINGZ (we only use 2d)")
+		FetchContent_Declare(Clipper2
+			GIT_REPOSITORY https://github.com/AngusJohnson/Clipper2.git
+			GIT_TAG ff378668baae3570e9d8070aa9eb339bdd5a6aba
+			GIT_PROGRESS TRUE
+			SOURCE_SUBDIR CPP
+			)
+		FetchContent_MakeAvailable(Clipper2)
+		if(NOT EMSCRIPTEN)
+			install(TARGETS Clipper2)
+		endif()
+	endif()
 endif()
-configure_file(manifold.pc.in ${CMAKE_CURRENT_BINARY_DIR}/manifold.pc @ONLY)
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/manifold.pc
-  DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
 
+# If we're supporting mesh I/O, we need assimp
+if(MANIFOLD_EXPORT)
+	find_package(assimp REQUIRED)
 endif()
 
+add_subdirectory(src)
+add_subdirectory(include)
+add_subdirectory(bindings)
+add_subdirectory(samples)
+add_subdirectory(test)
+add_subdirectory(extras)
+
 # configuration summary, idea from openscad
 # https://github.com/openscad/openscad/blob/master/cmake/Modules/info.cmake
 message(STATUS "====================================")
 message(STATUS "Manifold Build Configuration Summary")
 message(STATUS "====================================")
 message(STATUS " ")
 if (MXECROSS)
-  message(STATUS "Environment: MXE")
+	message(STATUS "Environment: MXE")
 elseif (APPLE)
-  message(STATUS "Environment: macOS")
+	message(STATUS "Environment: macOS")
```

Comment:
This is minor, but we generally do two-space indenting around here. Does clang-format handle CMake? For some reason ours has never been set up to run on these files, but it probably should be.

Reply:
Hmm.  Good question.  A quick check with stackoverflow turns up https://github.com/cheshirekow/cmake_format and https://github.com/BlankSpruce/gersemi but I've not tried them to know how they work.  For BRL-CAD we usually just add a footer to help the editors:  https://github.com/BRL-CAD/brlcad/blob/f8d7be0f755ff0f332d5c25143fa06635343d57c/CMakeLists.txt#L2419

Reply:
I committed an example in the pr reformatting the toplevel CMakeLists.txt the say way BRL-CAD does it - if that looks OK I can do the other CMake files similarly.